go-controller/cmd/ovnkube-identity/ovnkubeidentity.go

```diff
@@ -0,0 +1,377 @@
+	// on the socket to allow for two instances running at the same time (required during upgrades).
+	// The webhook server is set up and started in a very similar way to the default one:
+	// https://github.com/ovn-org/ovn-kubernetes/blob/7c0838bb46d6de202f509abe47609c8da09311b2/go-controller/vendor/sigs.k8s.io/controller-runtime/pkg/webhook/server.go#L212
+
+	client, err := kubernetes.NewForConfig(restCfg)
+	if err != nil {
+		return fmt.Errorf("error creating clientset: %v", err)
+	}
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	webhookMux := http.NewServeMux()
+
+	klog.Infof("Extra users are: %v", cliCfg.extraAllowedUsers.Value())
+	nodeWebhook := admission.WithCustomValidator(
+		scheme.Scheme,
+		&corev1.Node{},
+		ovnwebhook.NewNodeAdmissionWebhook(cliCfg.enableInterconnect, cliCfg.enableHybridOverlay, cliCfg.extraAllowedUsers.Value()...),
+	).WithRecoverPanic(true)
+
+	nodeHandler, err := admission.StandaloneWebhook(
+		nodeWebhook,
+		admission.StandaloneOptions{
+			Logger:      logger.WithName("node.network-identity"),
+			MetricsPath: "node.network-identity",
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to setup the node admission webhook: %w", err)
+	}
+	webhookMux.Handle("/node", nodeHandler)
+
+	// in non-ic ovnkube-node does not have the permissions to update pods
+	if cliCfg.enableInterconnect {
+		informerFactory := informers.NewSharedInformerFactory(client, 10*time.Minute)
+		nodeInformer := informerFactory.Core().V1().Nodes().Informer()
+		informerFactory.Start(stopCh)
+		klog.Infof("Waiting for caches to sync")
+		cache.WaitForCacheSync(c.Context.Done(), nodeInformer.HasSynced)
+
+		nodeLister := listers.NewNodeLister(nodeInformer.GetIndexer())
+		podWebhook := admission.WithCustomValidator(
+			scheme.Scheme,
+			&corev1.Pod{},
+			ovnwebhook.NewPodAdmissionWebhook(nodeLister, cliCfg.extraAllowedUsers.Value()...),
+		).WithRecoverPanic(true)
+		podHandler, err := admission.StandaloneWebhook(
+			podWebhook,
+			admission.StandaloneOptions{
+				Logger:      logger.WithName("pod.network-identity"),
+				MetricsPath: "pod.network-identity",
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("failed to setup the pod admission webhook: %w", err)
+		}
+		webhookMux.Handle("/pod", podHandler)
+	}
+
+	cfg := &tls.Config{
+		NextProtos: []string{"h2"},
+		MinVersion: tls.VersionTLS10,
+	}
+
+	certPath := filepath.Join(cliCfg.certDir, "tls.crt")
+	keyPath := filepath.Join(cliCfg.certDir, "tls.key")
+	certWatcher, err := certwatcher.New(certPath, keyPath)
+	if err != nil {
+		return fmt.Errorf("failed to setup certwatcher: %v", err)
+	}
+	cfg.GetCertificate = certWatcher.GetCertificate
+
+	go func() {
+		if err := certWatcher.Start(c.Context); err != nil {
+			klog.Errorf("Certificate watcher failed to start: %v", err)
+		}
+	}()
+	srv := &http.Server{
+		Handler:           webhookMux,
+		IdleTimeout:       90 * time.Second,
```

Comment:
what's an idletimeout?

Reply:
This is a parameter of the `http.Server` described here:
```
// IdleTimeout is the maximum amount of time to wait for the
// next request when keep-alives are enabled. If IdleTimeout
// is zero, the value of ReadTimeout is used. If both are
// zero, there is no timeout.
IdleTimeout time.Duration
```
As I mentioned here https://github.com/ovn-org/ovn-kubernetes/pull/3830/files/31b5e7c0417f72dcf8e13d10a8829b669d3e5d61..3fd8b1cb201bc8712010ea3d6fe0b73d324b2b2f#diff-b9f61853f3ebb69979208d961322bea4d37ce9531bddfdf1980bd5a7b7fe20e3R215-R218 this is based on the default webhook implementation which I had to adapt.